frappe/core/doctype/prepared_report/prepared_report.py

```diff
@@ -211,9 +214,9 @@ def expire_stalled_report():
 def delete_prepared_reports(reports):
 	reports = frappe.parse_json(reports)
 	for report in reports:
-		frappe.delete_doc(
-			"Prepared Report", report["name"], ignore_permissions=True, delete_permanently=True
-		)
+		prepared_report = frappe.get_doc("Prepared Report", report["name"])
+		if prepared_report.is_permitted():
```

Comment:
Prepared report already has a `has_permission` controller. We can just use it by calling `prepared_report.check_permission("read")` or something like that.

Reply:
Fixed :+1: